modules/core/src/ocl.cpp

```diff
@@ -2925,6 +2925,44 @@ CV_EXPORTS bool useSVM(UMatUsageFlags usageFlags)
 #endif // HAVE_OPENCL_SVM
 
 
+void attachContext(String& platformName, void* platformID, void* context, void* deviceID)
+{
+    cl_platform_id pl;
+    cl_device_id   d;
+
+    d = (cl_device_id)deviceID;
+
+    // get platform ID from user supplied device
+    CV_OclDbgAssert(clGetDeviceInfo(d, CL_DEVICE_PLATFORM, sizeof(cl_platform_id), &pl, NULL) == CL_SUCCESS);
```

Comment:
If there is something wrong with OpenCL runtimes (you assume it and try to check this by "platformName"), then you may receive exception/crash during execution of this function, because pointer `d` is alien for OpenCV OpenCL runtime.
`CV_OclDbgAssert` is misused here. Errors should be raised anyway.
You didn't initialize `pl` variable, so if you bypass this error then value of this variable is undefined for code below.

I still believe that `initializeContextFromHandle` is enough. If you want to guard from mismatched OpenCL dlls you may try to use `GetModuleHangle("OpenCL.dll)` near `LoadLibrary` to bypass it (and you can use library same as user - in case of lazy OpenCL initialization).